File: pset6/caesar.py
import sys
import cs50

def main():
    if (len(sys.argv)!=2):
        print("error")
        exit(1)
        
    k=int(sys.argv[1])
    k=k%26
    print("plaintext:",end="")
    c=cs50.get_string()
    print("ciphertext:",end="")
    l=len(c)
    for i in range(l):
        b=ord(c[i])
        if(b>=65 and b<=90):
            if(b+k>90):
                b=b+k-26
            else:
                b=b+k
        elif(b>=97 and b<=122):
            if(b+k>122):
                b=b+k-26
            else:
                b=b+k
        print("{}".format(chr(b)),end="")
    print("")
    exit(0)
            
if __name__ == "__main__":
    main()
        

    
    
    
        
            
        

File: pset2/caesar/caesar.c
#include<stdio.h>
#include<ctype.h>
#include<string.h>
#include<stdlib.h>
#include<cs50.h>
int main(int argc,string argv[])
{
    if(argc!=2)
    {
        printf("error");
    return 1;
    }
    int k=atoi(argv[1]);
    k=k%26;
    printf("plaintext:");
    string c=get_string();
    printf("ciphertext:");
    int l=strlen(c);
    for(int i=0;i<l;i++)
    {
        if(c[i]>=65&&c[i]<=90)
        {
            if(c[i]+k>90)
            c[i]=c[i]+k-26;
            else
            c[i]=c[i]+k;
        }
        else if(c[i]>=97&&c[i]<=122)
        {
            if(c[i]+k>122)
            c[i]=c[i]+k-26;
            else
            c[i]=c[i]+k;
        }
        printf("%c",c[i]);
    }
    printf("\n");
    return 0;
}